comp/vision/gamepiece_finder24.py

```diff
@@ -0,0 +1,205 @@
+from enum import Enum
+import cv2
+import libcamera
+import msgpack
+import numpy as np
+import time
+
+from cscore import CameraServer
+from ntcore import NetworkTableInstance
+from picamera2 import Picamera2
+from pupil_apriltags import Detector
+from wpimath.geometry import Translation2d
+
+import math
+
+class NotePosition:
+    pose: Translation2d
+
+class GamePieceFinder:
+
+    def __init__(self, topic_name, camera_params):
+        self.object_lower = (0 , 0, 200)
+        self.object_higher = (255, 170, 255)
+        self.scale_factor = 1
+        self.width = camera_params[0]
+        self.height = camera_params[1]
+        self.object_height = .105
+        self.theta = 0
+        self.topic_name = topic_name
+        self.initialize_nt()    
+        self.output_stream = CameraServer.putVideo("Processed", self.width, self.height)
+
+    def initialize_nt(self):
+        """Start NetworkTables with Rio as server, set up publisher."""
+        self.inst = NetworkTableInstance.getDefault()
+        self.inst.startClient4("retro-finder")
+        # this is always the RIO IP address; set a matching static IP on your
+        # laptop if you're using this in simulation.
+        self.inst.setServer("10.1.0.2")
+        # Table for vision output information
+        topic_name = "vision/" + self.serial
+        self.vision_fps = self.inst.getDoubleTopic(topic_name + "/fps").publish()
+        self.vision_latency = self.inst.getDoubleTopic(
+            topic_name + "/latency"
+        ).publish()
+
+        # work around https://github.com/robotpy/mostrobotpy/issues/60
+        self.inst.getStructTopic("bugfix", NotePosition).publish().set(
+            NotePosition(Translation2d())
+        )
+
+        # blip array topic
+        self.vision_nt_struct = self.inst.getStructArrayTopic(
+            topic_name + "/NotePositions", NotePosition
```

Comment:
make the key match java.  here you use "Notepositions" and in java you use "Noteposition"